### DISCUSSION TITLE
requestPurchase({sku: 'validSku'}) not completing or failing

### DISCUSSION THREAD

**Discussion by mjfre** [hyochan/react-native-iap#2253] [General]
**Description**

I switched to using react-native-iap hooks, but when I call requestPurchase({sku: 'validSku'}) the request does not proceed to either the 'then' or the 'catch', and the apple confirm purchase dialog does not appear.  I have followed the [hooks walkthrough](https://react-native-iap.dooboolab.com/docs/api-reference/hooks/), and went through every piece of documentation/issue/discussion I can find, to no avail.  I provided the withIapContext at the app-level.  Within the same component where I am experiencing the issue, I am **successfully** calling getProducts(...) and getSubscriptions(...) in a useEffect() that executes when the component mounts.   I have also confirmed that connected === true immediately before requestPurchase, and that the value provided as a parameter to requestPurchase is as expected.  I previously had all in-app purchases and subscriptions functioning completely, including backend receipt validation, so it doesn't seems as though there was in issue setting up the products on the apple side.  In fact, when I provide an invalid sku to requestPurchase(), I receive an error, so it seems it is doing something.

**Expected Behavior**

When requestPurchase({sku: 'validSku'}) is called, the call resolves and the Apple purchase dialog appears.  

**Code Snippets**

From App.js
```
import {withIAPContext} from "react-native-iap";  

...

export default withIAPContext(App);
```

From PurchaseView.js
```
import {useIAP} from "react-native-iap";

...

//START OF PART THAT WORKS FINE

const {
        connected,
        products,
        subscriptions,
        getProducts,
        getSubscriptions,
        finishTransaction,
        currentPurchase,
        currentPurchaseError,
        requestPurchase,
        requestPurchaseWithQuantityIOS,
        clearTransactionIOS
    } = useIAP();

    useEffect(() => {
        getSubscriptions({skus: premiumSubscriptionItemNames})
            .then((result) => {
                ...
            })
            .catch(() => {
               ...
            });
        getProducts({skus: consumableItemNames})
            .then((result) => {
              ...
            })
            .catch(() => {
                ...
            });

    }, []);

//END OF PART THAT WORKS FINE

...

const handlePurchase = async (sku) => {

        alert(sku + ' ' + connected);                  //output: '700credits' true

        await requestPurchase({sku: sku})
            .then((result) => {
                alert('transaction requested')       //this never runs
            })
            .catch((response) => {
                alert('request purchase failed')     //this also never runs
            });
};

...

<Pressable
        style={styles.PurchaseOption}
        disabled={consumable700CreditLocalizedPrice === null}
        onPress={() => {
        handlePurchase('700credits');
}}>
```

**Environment:**

- react-native-iap: 12.0.0 (also tried 11.0.7 as one person had luck downgrading) 
- react-native: 0.71.2
- Platforms (iOS, Android, emulator, simulator, device):  
      - iPhone 14 (iOS version 16.0.3)
      - Xcode simulator (I know I shouldn't use this for testing, but it used to at least display the apple purchase dialog) 

**To Reproduce**

1. Follow the directions in the [hooks walkthrough](https://react-native-iap.dooboolab.com/docs/api-reference/hooks/)
2. Call requestPurchase({sku: 'validSku'})  

---

**Additional Context**

Any help would be greatly appreciated!  This issue is currently blocking the release of my App :(

**Comment by andresesfm** [comment]
Have you tried the IapExample app part of this repo?
